src/v/net/tests/conn_quota_test.cc

```diff
@@ -93,12 +95,14 @@ struct conn_quota_fixture {
         scq
           .invoke_on(
             shard,
-            [shard, this, take_units, addr](conn_quota& cq) -> ss::future<> {
-                for (unsigned int j = 0; j < take_units; ++j) {
-                    auto u = co_await cq.get(addr);
+            [shard, this, take_units, addr](conn_quota& cq) {
+                auto range = boost::irange(0u, take_units);
+                return ss::do_for_each(range, [this, shard, addr, &cq](auto) {
```

Comment:
I think the lifetime of `range` is in question here. Maybe wrap it with a `do_with`

Reply:
Same comment for the other `do_for_each` invocations added in this change

Reply:
is it problematic?

Reply:
I feel like this should be fine -- the range is only used to count the number of tasks to schedule; the actual members of the range don't get used at all, right?

Reply:
i think Rob is concerned that do_for_each is holding onto a reference to the boost::irange object/container from the stack. it looks problematic, but i think boost::irange isn't really a container and the iterators can live on their own--which do_for_each moves from to keeps alive. there are several examples this pattern in seastar. it doesn't work for other types of containers.

Reply:
nice, didn't know that +1 LGTM